editor/src/clj/editor/app_view.clj

```diff
@@ -2011,26 +2011,45 @@ If you do not specifically require different script states, consider changing th
       (ui/on-closed! stage (fn [_] (dispose-scene-views! app-view)))
       app-view)))
 
+(defn- make-info-box! []
+  (let [info-panel (HBox.)
+        left-label (Label. "This file is part of a library and cannot be saved.  ")
+        right-link (Hyperlink. "Read more…")
+        spacer (Region.)]
+    (HBox/setHgrow spacer Priority/ALWAYS)
+    (.setStyle info-panel "-fx-background-color: #444; -fx-padding: 4;")
+    (.setTextFill left-label Color/LIGHTGRAY)
+    (.setTextFill right-link Color/CORNFLOWERBLUE)
+    (.setOnAction right-link (ui/event-handler _ (ui/open-url "https://www.defold.com/manuals/editor/")))
+    (.addAll  (.getChildren info-panel) (Arrays/asList (into-array Node [left-label spacer right-link])))
+    info-panel))
+
 (defn- make-tab! [app-view prefs workspace project resource resource-node
                   resource-type view-type make-view-fn ^ObservableList tabs
                   open-resource opts]
-  (let [parent     (AnchorPane.)
-        tab        (doto (Tab. (tab-title resource false))
-                     (.setContent parent)
-                     (.setTooltip (Tooltip. (or (resource/proj-path resource) "unknown")))
-                     (ui/user-data! ::view-type view-type))
+  (let [parent (AnchorPane.)
+        tab-content (VBox.)
+        content-nodes (if (resource/read-only? resource)
+                        [(make-info-box!) parent]
+                        [parent])
+        tab (doto (Tab. (tab-title resource false))
+              (.setContent tab-content)
```

Comment:
If the resource is not read-only, we don't need to wrap it in a VBox at all, so we could do something like:
```clj
tab-content (if (resource/readonly? resource) (vbox...) parent)
```

Reply:
Thx! Fixed!